test/SerializationTests.jl

```diff
@@ -1,14 +1,68 @@
 module SerializationTests
 
-using Base.Test
+using Compat
+using Compat.Test
 using BenchmarkTools
 
-old_data = BenchmarkTools.load(joinpath(dirname(@__FILE__), "data_pre_v006.jld"), "results")
-BenchmarkTools.save(joinpath(dirname(@__FILE__), "tmp.jld"), "results", old_data)
-new_data = BenchmarkTools.load(joinpath(dirname(@__FILE__), "tmp.jld"), "results")
+eq(x::T, y::T) where {T<:Union{BenchmarkTools.supported_types...}} =
+    all(i->eq(getfield(x, i), getfield(y, i)), 1:fieldcount(T))
+eq(x::T, y::T) where {T} = isapprox(x, y)
 
-@test old_data == new_data
+tmp = joinpath(@__DIR__, "tmp.json")
+isfile(tmp) && rm(tmp)
 
-rm(joinpath(dirname(@__FILE__), "tmp.jld"))
+@testset "Successful (de)serialization" begin
+    b = @benchmarkable sin(1)
+    tune!(b)
+    bb = run(b)
+
+    BenchmarkTools.save(tmp, b, bb)
+    @test isfile(tmp)
+    rm(tmp)
+
+    results = BenchmarkTools.load(tmp)
+    @test results isa Vector{Any}
+    @test length(results) == 2
+    @test eq(results[1], b)
+    @test eq(results[2], bb)
+end
+
+@testset "Deprecated behaviors" begin
+    @test_throws ArgumentError BenchmarkTools.save("x.jld", b)
+    @test_throws ArgumentError BenchmarkTools.save("x.txt", b)
+    @test_throws ArgumentError BenchmarkTools.save("x.json")
+    @test_throws ArgumentError BenchmarkTools.save("x.json", 1)
+
+    @test_warn "Naming variables" BenchmarkTools.save(tmp, "b", b)
+    @test isfile(tmp)
+    results = BenchmarkTools.load(tmp)
+    @test length(results) == 1
+    @test eq(results[1], b)
+    rm(tmp)
+
+    @test_throws ArgumentError BenchmarkTools.load("x.jld")
+    @test_throws ArgumentError BenchmarkTools.load("x.txt")
+    @test_throws ArgumentError BenchmarkTools.load("x.json", "b")
+end
+
+@testset "Error checking" begin
+    isfile(tmp) && rm(tmp)
+    open(tmp, "w") do f
+        print(f, """
+        {"never":1,[{"gonna":2,"give":3,"you":4,"up":5}]}
```

Comment:
Please consult [the manual](https://www.youtube.com/watch?v=dQw4w9WgXcQ) before adding a test like this.

Reply:
!!!!!!!!

Reply:
You're right Jarrett, my apologies. We're no strangers to unit testing; you know the rules and so do I.